### ISSUE TITLE
layers from manifest don't match image configuration

### ISSUE THREAD

**Issue by adamyi** [bazelbuild/rules_docker#1075]
I got `layers from manifest don't match image configuration` when pulling a container pushed with `container_push` to gcr. Might be related to https://github.com/docker/distribution/issues/1439

Repro:
```
WORKSPACE
new_container_pull(
    name = "tomcat-jython",
    registry = "index.docker.io",
    repository = "adamyi/tomcat-jython",
    tag = "latest",
)
BUILD
container_image(
    name = "image",
    base = "@tomcat-jython//image",
    visibility = ["//visibility:public"],
)
```

**Comment by adamyi** [comment]
This seems to be an issue with `new_container_pull`. I switched back to `container_pull` and it no longer triggers this error.

**Comment by smukherj1** [comment]
Hi @adamyi,

Thanks for opening this issue. `new_container_pull` is a new rule using a new Go backend and might have issues. We will take a look at why this is happening as this might be a bug in the new Go implementation. In the meantime, we suggest using `container_pull`. Keeping this issue open to track looking into why `new_container_pull` is unable to pull this image.

**Comment by smukherj1** [comment]
The fix for this is included in rules_docker release v0.10.0. Please let us know if that's not the case.
